src/noob/yaml.py

```diff
@@ -68,7 +76,14 @@ def to_yamls(self, **kwargs: Any) -> str:
             **kwargs: passed to :meth:`.BaseModel.model_dump`
         """
         data = self._dump_data(**kwargs)
-        return yaml.dump(data, Dumper=YamlDumper, sort_keys=False)
+        if hasattr(self, "_yaml_source") and self._yaml_source is not None:
+            self._yaml_source.update(data)
+            data = self._yaml_source
```

Comment:
update the yaml source with any new values from the model, preserving comments and etc.